### ISSUE TITLE
Remove the integrated "fast server" and move all web logic into the Puma server

### ISSUE THREAD

**Issue by adamcooke** [postalserver/postal#1427]
The fast server handles serving requests for click/open tracking. We call it the fast server because it's designed to respond to request quickly. It also helps handle issuing Let's Encrypt certificates for click/tracking domains. But...

* Writing our own web server is painful
* Managing our own Let's Encrypt integration is also painful
* Open/click tracking is evil 

In light of these things, rather than completely remove the option for click/open tracking (which is actually my preference and may happen one day!), I want to move it into the existing Puma server and delegate the SSL setup to the server administrator.

After this is complete, the built-in web server (that currently handles other Postal traffic) will be responsible for handling requests for click/open tracking endpoints. A server admin will need to configure their web proxy appropriately with SSL to route requests to that endpoint. This means that server admins will have an additional responsibility for setting these up in a way that suits them. 

By doing this we remove a significant burden from the project and will give us time to focus on other areas. I, for one, will not be sad to see the back of it.

**Comment by willpower232** [comment]
This makes sense to me and should be no trouble for our Glorious Dockerised Future.

**Comment by adamcooke** [comment]
Ah, the ol' GDF.

**Comment by nhadsall** [comment]
Wow, getting rid of open/click tracking would make postal nearly useless. All modern email systems have this feature. I’d be so bummed f we had to abandon postal after all the time we have invested to get it running.

**Comment by willpower232** [comment]
Click tracking will still be around but you'll have to handle your own routing for the URL, it wouldn't be as automatic as it is now.

The change makes the code much easier to maintain and the processes easier to run.

**Comment by Igcorreia** [comment]
@willpower232  as long as we can do it will be good. But like @nhadsall mentioned this was THE REASON to move to postal at the first place.

**Comment by nhadsall** [comment]
As in command line for each new url? What kind of “handle” are we talking about here?

**Comment by willpower232** [comment]
I believe users would have to create a vhost in nginx or whatever for each tracking domain and point it at the relevant local port which makes its way to Postal.

As part of this, users would also need to provide their own SSL certificate for each domain.

If click tracking is removed from Postal itself in the future, then so be it but as Postal is open sourced there is nothing stopping anyone restoring it and maintaining their own version of it with whatever customisations they desire. The Glorious Dockerised Future makes this even easier as builds are easier to distribute.

**Comment by adamcooke** [comment]
As I said, I don't have any plans to remove it but I don't agree with the concept of tracking people opening emails.

**Comment by Igcorreia** [comment]
@adamcooke I miss understood. Ony our case we don't track 99% of the emails, but in a specific project/server it was mandatory for the purpose of the project. Consider adding this optional either per domain, credentials, or server. Just a suggestion and I 100% agreed with you. 

ps: there is always someone that will use postal as an email newsletter platform so tracking may be mandatory

**Comment by nhadsall** [comment]
Tacking is mission critical for us and for clients. We have had several cases were deliverability is not enough. An open or link click has saved us when invoices went to spam. Both the sender AND the receiver were grateful that we saw it wasn't opened and took the necessary steps (a phone call) to intervene. I am sure there are a dozen use cases besides "spying on our marketing efforts" that warrant the option. I don't say this to argue, but as an attempt to gently persuade. @adamcooke As head of software for a project we rely on, your opinion does matter :) And I do disagree agree with @willpower232 that "nothing stopping anyone" isn't the case. You guys do heroic work maintaining this for FREE and there's no way a small agency like ours could put in the work you do. We have already invested a lot to get this running, and that's nothing compared to what you all have done. So I humbly disagree, but also have a ton of gratitude :)
